framework/src/utils/MooseUtils.C

```diff
@@ -31,10 +31,42 @@
 
 // System includes
 #include <sys/stat.h>
+#include <numeric>
 
 namespace MooseUtils
 {
 
+int
+levenshteinDist(const std::string & s1, const std::string & s2)
+{
+  // To change the type this function manipulates and returns, change
+  // the return type and the types of the two variables below.
+  auto s1len = s1.size();
+  auto s2len = s2.size();
+
+  auto column_start = (decltype(s1len))1;
```

Comment:
You should be able to right this line of code with decltype on the LHS:
`decltype(silen) column_start = 1;` 

The only time it needs to be on the RHS is when you are trying to drop the const. However in this case it looks like `column_start` is constant.